### ISSUE TITLE
 TensorRT loses defined shapes

### ISSUE THREAD

**Issue by rankeey** [tensorflow/tensorflow#25417]
System information
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
Bazel version: N/A
GPU model and memory: P4 8G
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
TensorFlow installed from (source or binary): docker
TensorFlow version (use command below): 1.12
Python version: 3.5
CUDA/cuDNN version: 10/7.4.1
Describe the problem

When tf.contrib.tensorrt compiles a graph which can be fully compiled it loses shape information.

https://github.com/tensorflow/tensorrt.git
just run the tensorrt/tftrt/examples/image-classification/image_classification.py with int8 parameter like the follow command and get the frozen model from graphs directory.

 python image_classification.py --model resnet_v1_50 --data_dir /data/ImageNetVal/ --precision fp32   --batch_size=1 --use_trt

the one frozen from tensorrt:
Tensor("input:0", shape=(?, 224, 224, 3), dtype=float32)
Tensor("logits:0", dtype=float32)

and the native one:

Tensor("input:0", shape=(?, 224, 224, 3), dtype=float32)
Tensor("logits:0", shape=(?, 1001), dtype=float32)

the difference is the shape in logits from tensorrt have been lost.

script as follow:

import tensorflow as tf
from tensorflow.contrib import tensorrt as trt
import os
import shutil
graph_pb = 'frozen_graph_resnet_v1_50_1_fp32_1.pb'

with tf.gfile.GFile(graph_pb, "rb") as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())
sigs = {}
with tf.Session(graph=tf.Graph()) as sess:
    for n in graph_def.node:
        if n.name == "logits":
            print(n)
        if n.name == "input":
            print(n)

@joeyearsley

**Comment by ymodak** [comment]
@trevor-m Can you PLTAL? Thanks!

**Comment by samikama** [comment]
@rankeey , It is not trivial to estimate the output shapes of TRTEngineOp for given inputs, so its shape inference is disabled as of now. What you print is output of shape inference. Your network will run fine, it just can't do shape inference to print the shapes.
